common/include/pcl/common/impl/centroid.hpp

```diff
@@ -114,8 +117,11 @@ pcl::compute3DCentroid (const pcl::PointCloud<PointT> &cloud,
       centroid[2] += cloud[i].z;
       ++cp;
     }
-    centroid /= static_cast<Scalar> (cp);
-    centroid[3] = 1;
+    if (cp != 0)
+    {
+      centroid /= static_cast<Scalar> (cp);
+      centroid[3] = 1;
+    }
```

Comment:
Notice that by the time we get here, the centroid variable has already been modified. To be fully compliant with the docstring, this accumulation needs to happen in a "proxy" variable whose value at the end gets written to centroid.